solcx/install.py

```diff
@@ -212,18 +224,44 @@ def get_installed_solc_versions(solcx_binary_path=None):
     )
 
 
+def _check_subprocess_call(command, message=None, verbose=False, **proc_kwargs):
+    if message:
+        LOGGER.debug(message)
+
+    command_str = " ".join(map(str, command))
+    LOGGER.info(f"Executing:  {command_str}")
+
+    return subprocess.check_call(
+        command, stderr=subprocess.STDOUT if verbose else subprocess.DEVNULL, **proc_kwargs
+    )
+
+
+def _get_temp_folder():
+    path = Path(tempfile.gettempdir()).joinpath(f"solcx-tmp-{os.getpid()}")
+    if path.exists():
+        shutil.rmtree(str(path))
+    path.mkdir()
+    return path
+
+
+# ----------------------------------------------------------------------------
+# ----------------------------------------------------------------------------
+"""
+The following functions download and install solc to your OS
+"""
+
+
 def install_solc(version, allow_osx=False, show_progress=False, solcx_binary_path=None):
-    platform = _get_platform()
     version = _check_version(version)
+    is_version_installed(version=version, solcx_binary_path=solcx_binary_path)
```

Comment:
I think the return value of this call should be checked, and if `True`, don't install.

This is probably why the test suite is taking so long to run..  in the CI it should complete in about a minute.

Reply:
Good find.  When I originally wrote this line I assumed `is_version_installed` would end the function if it returns `False`, when in reality `install_solc` continues to execute.